src/ripple/basics/PerfLog.h

```diff
@@ -0,0 +1,163 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2018 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_BASICS_PERFLOG_H
+#define RIPPLE_BASICS_PERFLOG_H
+
+#include <ripple/core/JobTypes.h>
+#include <boost/filesystem.hpp>
+#include <ripple/json/json_value.h>
+#include <chrono>
+#include <cstdint>
+#include <memory>
+#include <string>
+
+using namespace std::chrono;
+using steady_time_point = time_point<steady_clock>;
+using namespace std::chrono_literals;
```

Comment:
Hi Mark, I'm not saving this comment for later because it's really important.  `using` statements have scope.  You've put these `using` statements at global scope.  That means the `using` statements you've put here effect every file that includes this header file (through however many levels of nested includes).  That's considered bad form.  Human readers of other files can't see that they've been infected by these `using` declarations, but it can change the way those files compile.  In the very best case it will cause strange error messages that can be very hard to track down.

My suggestion would be to move these `using` declarations inside of your base class, `PerfLog`.  If that's too constraining, then the largest scope I would recommend giving them would be inside the `perf` namespace.  But try moving them to the top inside of `PerfLog` and see if that gets you far enough before deciding to give them larger scope.

Thanks.

Reply:
Just to be extra clear, I'm suggesting you remove these three `using` declarations.  Then do something like this at the top of PerfLog:
```
class PerfLog
{
public:
     using steady_time_point = std::chrono::time_point<std::chrono::steady_clock>;
     using seconds = std::chrono::seconds;
     using microseconds = std::chrono::microseconds;

    /**
     * Configuration from [perf] section of rippled.cfg.
     */
    struct Setup
    {
        boost::filesystem::path perf_log;
        seconds log_interval {1};
    };
    ...
```